extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -590,6 +584,21 @@ public void build(
                                 hibernateValidatorBuildTimeConfig)));
     }
 
+    private static Set<Class<?>> toClassSet(Set<DotName> dotNameSet) {
+        Set<Class<?>> classSet = new HashSet<>(dotNameSet.size());
+        for (DotName className : dotNameSet) {
+            try {
+                Class<?> type = Class.forName(className.toString(), true, Thread.currentThread().getContextClassLoader());
```

Comment:
That's not a good idea because the class can be available at deployment and not available at runtime, given our deployment classpath has a lot more dependencies.

I recommend the use of `QuarkusClassLoader.isClassPresentAtRuntime()` which was introduced for this particular use case.